src/go/transform-sdk/internal/cache/cache.go

```diff
@@ -0,0 +1,60 @@
+package cache
+
+import "math"
+
+type (
+	entry[V any] struct {
+		value           V
+		insertionNumber int
+	}
+	// A cache that evicts based on number of entries
+	Cache[K comparable, V any] struct {
+		underlying                 map[K]entry[V]
+		latestEntryInsertionNumber int
+		maxEntries                 int
+	}
+)
+
+// New returns a new cache with a limited set of entries.
+func New[K comparable, V any](maxSize int) Cache[K, V] {
+	return Cache[K, V]{
+		underlying:                 make(map[K]entry[V]),
+		latestEntryInsertionNumber: 0,
+		maxEntries:                 maxSize,
+	}
+}
+
+// Put adds an entry into the cache
+func (c *Cache[K, V]) Put(k K, v V) {
+	c.underlying[k] = entry[V]{value: v, insertionNumber: c.latestEntryInsertionNumber}
```

Comment:
This temporarily inserts one more than the limit, which means if you're really unlucky, the number of buckets could be twice what's required.

Reply:
Nice catch - fixed.